src/main/java/uk/gov/hmcts/reform/slc/services/steps/maptoletter/LetterMapper.java

```diff
@@ -15,16 +19,19 @@
 public class LetterMapper {
 
     private static final ObjectMapper objectMapper = new ObjectMapper();
+    private static final Validator validator = Validation.buildDefaultValidatorFactory().getValidator();
 
     public Letter from(IMessage msg) {
         try {
             Letter letter = objectMapper.readValue(msg.getBody(), Letter.class);
-            if (letter != null) {
+            Set<ConstraintViolation<Letter>> violations = validator.validate(letter);
```

Comment:
why do we need to validate this.Letter is validated before persisting into queue.

Reply:
There is no such thing as `letter == null` when you unmarshall the letter. It's either exception throw or letter object with null fields inside. So actually we need to test all the fields of null of letter object and not the object being null itself.

Hope that makes sense.

Reply:
I just checked and it throws `IllegalArgumentException`, which is not handled here

Reply:
😭